### ISSUE TITLE
expand_events duplicates events

### ISSUE THREAD

**Issue by petschki** [plone/plone.app.event#348]
Since https://github.com/plone/plone.app.event/pull/340 there are duplicated events in our event listing in the following scenario:

If you create a Collection (or in our case a contentlisting tile) and select `event_listing` to get expanded occurrences the original events are always shown additionally to the occurrences.

**Comment by petschki** [comment]
I'm sorry but I don't understand this logic:

https://github.com/plone/plone.app.event/blob/23f64443928fea1d0c322019a987913647a98850/plone/app/event/base.py#L285-L287

why should `expand_events` add the event object if its not within the given start/end range?

If you have a catalog query with `end >= now` your result returns the future event, but if the expand_event further limits the `end` range, the original event should be shown in the list if its after this limit ... or am I something missing?

In our project https://www.rankweil.at/veranstaltungen/ you can see the recurring event "Rankler Wochenmark" which is doubled the first time when it occurs (first is occurrence, second is original event) ... I I remove this one:

when I remove those lines above, everything works like it should ..

/cc @thet @frisi

**Comment by frisi** [comment]
the reasoning is described in https://github.com/plone/plone.app.event/issues/261:
> usecase: event-index of a portal. we limit recurrences to today+365days, but want to show events happening next year too.

iiuc the following condition should omit the event if it is in the list already.
can you check why this is not the case for you? mabe this can be fixed by adapting the condition.

```
         if obj not in occ_list: 
             occ_list.append(obj) 
```

**Comment by petschki** [comment]
Thank you for the explanation, but this in fact is exactly our problem. If you have a Collection, the `event_listing` only returns the results from `expand_events`:

https://github.com/plone/plone.app.event/blob/3f785479c32ba4bff9bdeadf488aa099d53894f3/plone/app/event/browser/event_listing.py#L164-L175

Now if you limit your `end` in the collection query, you also get events after this limit. So for my understanding 
`expand_events` should filter out everything which isn't within the given boundaries.

I'll try to create a PR which fixes our issue by keeping your condition above.

**Comment by petschki** [comment]
Hm ... I came a little closer to our problem (which didn't really have something to do with the "future event" problem). Strangely the original event was added to the first occurrcence of this event allthough the event start is in the past. Now this is because `IEventAccessor(obj).start` returns the "upcoming event" within the occurrences see here:

https://github.com/plone/plone.app.event/blob/3f785479c32ba4bff9bdeadf488aa099d53894f3/plone/app/event/dx/behaviors.py#L484-L493

So the original event is always added because `(IEventAccessor(obj).start >= (end or start))` is always true.
